sdk/cosmos/azure_data_cosmos/tests/cosmos_items.rs

```diff
@@ -405,3 +407,341 @@ pub async fn item_null_partition_key(context: TestContext) -> Result<(), Box<dyn
     account.cleanup().await?;
     Ok(())
 }
+
+#[recorded::test]
+pub async fn item_replace_if_match_etag(context: TestContext) -> Result<(), Box<dyn Error>> {
+    let account = TestAccount::from_env(context, None).await?;
+    let cosmos_client = account.connect_with_key(None)?;
+    let container_client = create_container(&account, &cosmos_client).await?;
+
+    //Create an item
+    let mut item = TestItem {
+        id: "Item1".into(),
+        partition_key: Some("Partition1".into()),
+        value: 42,
+        nested: NestedItem {
+            nested_value: "Nested".into(),
+        },
+        bool_value: true,
+    };
+
+    let response = container_client
+        .create_item("Partition1", &item, None)
+        .await?;
+
+    //Store Etag from response
+    const ETAG: HeaderName = HeaderName::from_static("etag");
+    let etag = response
+        .headers()
+        .get_str(&ETAG)
+        .ok()
+        .map(|s| Etag::from(s.to_string()));
+
+    //Replace item with correct Etag
+    item.value = 24;
+    item.nested.nested_value = "Updated".into();
+
+    //could change response name to replaced item for better understanding
+    let response = container_client
+        .replace_item(
+            "Partition1",
+            "Item1",
+            &item,
+            Some(ItemOptions {
+                if_match_etag: etag,
+                enable_content_response_on_write: false,
+                ..Default::default()
+            }),
+        )
+        .await?;
+
+    let body = response.into_raw_body().collect_string().await?;
+    assert_eq!("", body);
+
+    //Replace item with incorrect Etag
+    item.value = 52;
+    item.nested.nested_value = "UpdatedAgain".into();
+
+    let response = container_client
+        .replace_item(
+            "Partition1",
+            "Item1",
+            &item,
+            Some(ItemOptions {
+                if_match_etag: Some("incorrectEtag".into()),
+                enable_content_response_on_write: true,
+                ..Default::default()
+            }),
+        )
+        .await;
+
+    match response {
+        Ok(_) => {
+            return Err(
+                "expected a 412 Precondition Failed error when using an incorrect ETag".into(),
+            );
+        }
+        Err(err) => {
+            assert_eq!(
+                Some(azure_core::http::StatusCode::PreconditionFailed),
+                err.http_status()
+            );
+        }
+    }
```

Comment:
This is _good_ error handling for product code, since you're properly `match`ing on the `Result<...>` returned by `.replace_item`. But in a test, where we _know_ that the server is supposed to return an error and the test should fail if it does not return what we expect, we can simplify this quite a bit!

 ```suggestion
    assert_eq!(
        Some(azure_core::http::StatusCode::PreconditionFailed),
        response.expect_err("expected the server to return an error").http_status()
    );
```

The `.expect_err` method works a lot like `.expect`. It's just the opposite. Where `.expect` is used to unpack the successful return value from a `Result` (and panic if the `Result` represents an error), `.expect_err` unpacks the _error_ from a `Result` (and panics if the `Result` represents a successful operation).